harness/go/harness/httpproxy.go

```diff
@@ -105,6 +98,20 @@ func (h *HTTPConnectProxyServer) handler(w http.ResponseWriter, r *http.Request)
 	} else {
 		h.UnauthedConnectionsTunneled.Add(1)
 	}
+
+	// Node's HTTP2 client is sensible to the order of packets at the very start of the connection.
+	// That is, if the client receives the server's first HTTP2 packet before it sends its own first
+	// HTTP2 packet, it will terminate the connection (HTTP2 GOAWAY) and fail. Therefore, wait for
+	// the first client packet before falling into full duplex mode.
+	buf := make([]byte, 32 * 1024)
+	readBytes, err := clientConn.Read(buf)
+	if err != nil {
+		panic(fmt.Sprintf("Expected client to send a first packet: %v", err))
+	}
+	if _, err := targetConn.Write(buf[:readBytes]); err != nil {
+		panic(fmt.Sprintf("Sending client's first packet to server failed: %v", err))
+	}
```

Comment:
Interesting, so server was sending something before the client was?

Reply:
Exact. In the image above, see packets 19 and 21.

Now, AFAIU, that's technically legit, and it certainly looks like most HTTP2 clients don't stumble on that. But I suppose this would be a rare occurence in a normal, non-proxied HTTP2 session, as the client will send its first HTTP2 packet immediately after completing the TCP handshake (i.e. after the ACK), whereas the server would only send its first packet after receiving that ACK. Even in a TCP Fast Open scenario, the delay between the client's receiving the handshake ACK and receiving the first data packet would very likely be enough for the client-side HTTP2 library to complete its internal initialization and write out its first packet.

Anyway, I'm opening bug tickets in upstream projects regarding this.